lib/segment/src/index/field_index/btree_index.rs

```diff
@@ -29,15 +33,31 @@ impl KeyEncoder for FloatPayloadType {
     }
 }
 
-#[derive(Default)]
 struct NumericIndex<T: KeyEncoder> {
     map: BTreeMap<Vec<u8>, u32>,
+    db: Arc<AtomicRefCell<DB>>,
+    store_cf_name: String,
     phantom: PhantomData<T>,
 }
 
 impl<T: KeyEncoder> NumericIndex<T> {
+    #[allow(dead_code)] // TODO(gvelo): remove when this index is integrated in `StructPayloadIndex`
+    pub fn new(db: Arc<AtomicRefCell<DB>>, store_cf_name: String) -> Self {
+        Self {
+            map: BTreeMap::new(),
+            db,
+            store_cf_name,
+            phantom: PhantomData::default(),
+        }
+    }
+
     pub fn add_value(&mut self, id: PointOffsetType, value: T) {
         let key = value.encode_key(id);
+        let db_ref = self.db.borrow();
+        let cf_handle = db_ref.cf_handle(&self.store_cf_name).unwrap();
+        db_ref
+            .put_cf_opt(cf_handle, &key, id.to_be_bytes(), &db_write_options())
+            .unwrap();
```

Comment:
@IvanPleshkov I agree with you , panicking on the write path is not a good idea, but, since this change involves updating of some `fn` on the `ValueIndexer` trait, I would prefer to make the return value refactor when we integrate all the new index implementations in `StructPayloadIndex` otherwise we will need to update the old index implementation where the `ValueIndexer` functions are infallible because they don't touch disk. What do you think about this approach ?

Reply:
I would rather just keep a TODO comment here, to not forget after